From 921d92d6027467adc15ca2b8584849e3f4f3f9b7 Mon Sep 17 00:00:00 2001
From: Eli Nidam <elini@marvell.com>
Date: Thu, 4 Jul 2013 16:26:07 +0300
Subject: [PATCH 0765/1825] neta: fix: Add subrutine to disable SMI PHY
 polling

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit 75f6986be47877f43a1311259b4a48f3c4d7f66a

	add mvNetaPhyAddrPollingDisable subrutine for disable SMI PHY polling

Change-Id: I1f6b583a563ac6e3e5d536f3c06c11d5a777b3e9
Signed-off-by: Eli Nidam <elini@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/2432
Reviewed-by: Dmitri Epshtein <dima@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/plat-armada/mv_hal/neta/gbe/mvNeta.c |   32 +++++++++++++++++++++---
 arch/arm/plat-armada/mv_hal/neta/gbe/mvNeta.h |    1 +
 2 files changed, 29 insertions(+), 4 deletions(-)

diff --git a/arch/arm/plat-armada/mv_hal/neta/gbe/mvNeta.c b/arch/arm/plat-armada/mv_hal/neta/gbe/mvNeta.c
index 515519f..2ef74c0 100644
--- a/arch/arm/plat-armada/mv_hal/neta/gbe/mvNeta.c
+++ b/arch/arm/plat-armada/mv_hal/neta/gbe/mvNeta.c
@@ -2090,10 +2090,34 @@ void mvNetaPhyAddrSet(int port, int phyAddr)
 
 	MV_REG_WRITE(ETH_PHY_ADDR_REG(port), regData);
 
-    /* Enable PHY polling */
-    regData = MV_REG_READ(ETH_UNIT_CONTROL_REG(port));
-    regData |= ETH_PHY_POLLING_ENABLE_MASK;
-    MV_REG_WRITE(ETH_UNIT_CONTROL_REG(port), regData);
+	/* Enable PHY polling */
+	regData = MV_REG_READ(ETH_UNIT_CONTROL_REG(port));
+	regData |= ETH_PHY_POLLING_ENABLE_MASK;
+	MV_REG_WRITE(ETH_UNIT_CONTROL_REG(port), regData);
+
+	return;
+}
+/*******************************************************************************
+* mvNetaPhyAddrPollingDisable - disable PHY polling
+*
+* DESCRIPTION:
+*       This routine diable PHY polling
+*
+* INPUT:
+*       int     portNo		- Port number.
+*
+* RETURN:
+*       None.
+*
+*******************************************************************************/
+void mvNetaPhyAddrPollingDisable(int port)
+{
+	unsigned int regData;
+
+	/* Enable PHY polling */
+	regData = MV_REG_READ(ETH_UNIT_CONTROL_REG(port));
+	regData &= ~ETH_PHY_POLLING_ENABLE_MASK;
+	MV_REG_WRITE(ETH_UNIT_CONTROL_REG(port), regData);
 
 	return;
 }
diff --git a/arch/arm/plat-armada/mv_hal/neta/gbe/mvNeta.h b/arch/arm/plat-armada/mv_hal/neta/gbe/mvNeta.h
index 58643a7..0bba6d3 100644
--- a/arch/arm/plat-armada/mv_hal/neta/gbe/mvNeta.h
+++ b/arch/arm/plat-armada/mv_hal/neta/gbe/mvNeta.h
@@ -719,6 +719,7 @@ MV_STATUS	mvNetaBpduRxq(int port, int rxq);
 
 void 		mvNetaPhyAddrSet(int port, int phyAddr);
 int 		mvNetaPhyAddrGet(int port);
+void		mvNetaPhyAddrPollingDisable(int port);
 
 void 		mvNetaPortPowerDown(int port);
 void 		mvNetaPortPowerUp(int port, MV_BOOL isSgmii, MV_BOOL isRgmii);
-- 
1.7.5.4

